### PR TITLE
BUG: overriden methods of subclasses of Styler are not called during rendering #52728

### LINKED ISSUE #52728: BUG: overriden methods of subclasses of Styler are not called during rendering
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas
from pandas.io.formats.style import Styler

class CustomStyler(Styler):
    def _translate(self, *args, **kwargs):  # override internal _translate method
        result = super()._translate(*args, **kwargs)
        raise NotImplementedError('TODO')

data = pandas.DataFrame({'column A': [4,3,7]})
CustomStyler(data).to_html()  # CustomStyler._translate never gets called

type(CustomStyler(data)._copy())  # returns Styler not CustomStyler
```

### Issue Description

rendering behaviour of Styler can't be customised by users, because internally Styler.render methods use ._copy to get a copy of the Styler that some internal changes are then applied to

This could be worked around by turning ._copy into a classmethod e.g. Styler._copy(cls, self) which returns a copy with the same type as self

### Expected Behavior

type(CustomStyler(data)._copy())  == CustomStyler

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 478d340667831908b5b4bf09a2787a11a14560c9
python           : 3.8.10.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.15.0-69-generic
Version          : #76~20.04.1-Ubuntu SMP Mon Mar 20 15:54:19 UTC 2023
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : en_GB.UTF-8

pandas           : 2.0.0
numpy            : 1.22.0
pytz             : 2021.3
dateutil         : 2.8.2
setuptools       : 56.0.0
pip              : 21.1.1
Cython           : None
pytest           : 6.2.5
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : 3.0.2
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.3
IPython          : 7.31.0
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.9
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : 2.0.1
zstandard        : None
tzdata           : 2023.3
qtpy             : None
pyqt5            : None

</details>

### PR DESCRIPTION
**Pull request by nicklausroach**
- [x] closes #52728 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -1554,7 +1558,8 @@ def _update_ctx_header(self, attrs: DataFrame, axis: AxisInt) -> None:
                 else:
                     self.ctx_columns[(j, i)].extend(css_list)
 
-    def _copy(self, deepcopy: bool = False) -> Styler:
+    @classmethod
+    def _copy(cls, self, deepcopy: bool = False) -> Styler:
```
Comment (attack68) [inline pandas/io/formats/style.py:1562]: Is it necessary to make this a class method? Although this is really only an internal method doing this completely changes its signature. Is it not sufficient to keep it as a default method and use only the `type(self)(..)` line below?

Comment (nicklausroach) [reply pandas/io/formats/style.py]: Totally agree, I think I was just going off of the suggestion in the issue, but that makes total sense. Made the change, also makes the diff much cleaner!

**Review by attack68 — APPROVED** [review]
LGTM thanks

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @nicklausroach

**Pull request merged**